docs/reference/modules/discovery/quorums.asciidoc

```diff
@@ -0,0 +1,181 @@
+This means that if there are three or four master-eligible nodes then the
+cluster can tolerate one of them being unavailable; if there are two or fewer
+master-eligible nodes then they must all remain available.
+
+After a node has joined or left the cluster the elected master must issue a
+cluster-state update that adjusts the voting configuration to match, and this
+can take a short time to complete. It is important to wait for this adjustment
+to complete before removing more nodes from the cluster.
+
+[float]
+==== Setting the initial quorum
+
+When a brand-new cluster starts up for the first time, one of the tasks it must
+perform is to elect its first master node, for which it needs to know the set
+of master-eligible nodes whose votes should count in this first election. This
+initial voting configuration is known as the _bootstrap configuration_ and is
+set in the <<modules-discovery-bootstrap-cluster,cluster bootstrapping
+process>>.
+
+It is important that the bootstrap configuration identifies exactly which nodes
+should vote in the first election, and it is not sufficient to configure each
+node with an expectation of how many nodes there should be in the cluster. It
+is also important to note that the bootstrap configuration must come from
+outside the cluster: there is no safe way for the cluster to determine the
+bootstrap configuration correctly on its own.
+
+If the bootstrap configuration is not set correctly then there is a risk when
+starting up a brand-new cluster is that you accidentally form two separate
+clusters instead of one. This could lead to data loss: you might start using
+both clusters before noticing that anything had gone wrong, and it will then be
+impossible to merge them together later.
+
+NOTE: To illustrate the problem with configuring each node to expect a certain
+cluster size, imagine starting up a three-node cluster in which each node knows
+that it is going to be part of a three-node cluster. A majority of three nodes
+is two, so normally the first two nodes to discover each other will form a
+cluster and the third node will join them a short time later. However, imagine
+that four nodes were erroneously started instead of three: in this case there
+are enough nodes to form two separate clusters. Of course if each node is
+started manually then it's unlikely that too many nodes are started, but it's
+certainly possible to get into this situation if using a more automated
+orchestrator, particularly if the orchestrator is not resilient to failures
+such as network partitions.
+
+The initial quorum is only required the very first time a whole cluster starts
+up: new nodes joining an established cluster can safely obtain all the
+information they need from the elected master, and nodes that have previously
+been part of a cluster will have stored to disk all the information required
+when restarting.
+
+[float]
+==== Cluster maintenance, rolling restarts and migrations
```

Comment:
This seems like it could be a separate page under "Discovery and cluster formation".